### PR TITLE
refactor: made cloneNode partially synchronous

### PR DESCRIPTION
**Pull request by istaiti**

### CODE REVIEW

**Comment by istaiti** [comment]
Current dependencies on/for this PR:
* master
  * **PR #9** <a href="https://app.graphite.dev/github/pr/inscreen/html-to-image/9" target="_blank">Graphite</a> 
    * **PR #10** <a href="https://app.graphite.dev/github/pr/inscreen/html-to-image/10" target="_blank">Graphite</a> 
      * **PR #8** <a href="https://app.graphite.dev/github/pr/inscreen/html-to-image/8" target="_blank">Graphite</a>  👈
        * **PR #11** <a href="https://app.graphite.dev/github/pr/inscreen/html-to-image/11" target="_blank">Graphite</a> 
          * **PR #12** <a href="https://app.graphite.dev/github/pr/inscreen/html-to-image/12" target="_blank">Graphite</a> 

This comment was auto-generated by [Graphite](https://app.graphite.dev/github/pr/inscreen/html-to-image/8?utm_source=stack-comment).

File: package.json [hunk 1]
```diff
@@ -1,6 +1,6 @@
 {
   "name": "@inscreen/html-to-image",
-  "version": "1.11.12",
+  "version": "1.12.0",
```
Comment (ramonsnir) [inline package.json]: Did they also release `1.12.0`? Let's not give random numbers to versions that conflict with _their_ versions, if we continue to incorporate changes from them. I'm pretty sure npm allows specifying more than three numbers in the version (e.g. `-is2`, `-is3`, or something else).

File: src/clone-node.ts [hunk 2]
```diff
@@ -134,8 +135,9 @@ function cloneCSSStyle<T extends HTMLElement>(nativeNode: T, clonedNode: T) {
         value = 'block'
       }
 
-      if (name === 'd' && clonedNode.getAttribute('d')) {
-        value = `path(${clonedNode.getAttribute('d')})`
+      let dAttribute
+      if (name === 'd' && (dAttribute = clonedNode.getAttribute('d'))) {
+        value = `path(${dAttribute})`
```
Comment (ramonsnir) [inline src/clone-node.ts]: What does this change do differently?

File: src/clone-node.ts [hunk 3]
```diff
@@ -192,77 +194,97 @@ function cloneScrollPosition<T extends HTMLElement>(
...
-
-  return clonedNode
-}
-
-async function ensureSVGSymbols<T extends HTMLElement>(
-  clone: T,
+// TODO: optimise this and run it as a "filter" - once per <use> tag
+// https://github.com/bubkoo/html-to-image/issues/341
+// currently - it runs `clone.querySelectorAll('use')` on each node and `document.querySelector(id)` for each tag found
+// async function ensureSVGSymbols<T extends HTMLElement>(
```
Comment (ramonsnir) [inline src/clone-node.ts]: What is this dead code doing? Don't leave commented-out code that you deleted in the codebase, we have version control to find old versions of the code.

**Review by ramonsnir — COMMENTED** [review]
There's no way to tell from the diff what you changed. Please summarize in text.

**Comment by istaiti** [comment]
@ramonsnir Got rid of the main async chain in cloneNode, by using `await`...
and then added two if conditions to make sure we're breaking the recursion on non html elements

File: src/clone-node.ts [hunk 4]
```diff
@@ -192,35 +193,50 @@ function cloneScrollPosition<T extends HTMLElement>(
...
+  }
+
+  if (!isElementNode(node)) {
+    return node.cloneNode(false) as HTMLElement
+  }
+
+  if (isRoot) {
+    // keeping for future improvements
   }
+  const style = window.getComputedStyle(node)
```
Comment (ramonsnir) [inline src/clone-node.ts]: Isn't this very slow?

Comment (istaiti) [reply src/clone-node.ts]: it is.. on my list for the next improvements

**Review by ramonsnir — APPROVED** [review]

**Comment by istaiti** [comment]
✅ This pull request merged successfully as part of a Graphite job
Job ID: Mudo7G0eRYbNUVVj4Ct1.
See details on [graphite.dev](https://app.graphite.dev/github/pr/inscreen/html-to-image/8)

**Pull request merged**